Jenkinsfile

```diff
@@ -11,109 +11,192 @@ pipeline {
+                                        sh 'mvn install -B -Dmaven.test.redirectTestOutputToFile=true -pl $ITESTS -nsu'
+                                    }
+                                }
                             }
-                            timeout(time: 1, unit: 'HOURS') {
-                                withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings') {
-                                    sh 'mvn clean install -B -T 1C -pl !$ITESTS'
-                                    sh 'mvn install -B -Dmaven.test.redirectTestOutputToFile=true -pl $ITESTS -nsu'
+                        },
+                        windows: {
+                            node('proxmox-windows') {
+                                bat 'git config --system core.longpaths true'
+                                retry(3) {
+                                    checkout scm
+                                }
+                                timeout(time: 1, unit: 'HOURS') {
+                                    withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS}', options: [artifactsPublisher(disabled: true), dependenciesFingerprintPublisher(disabled: true, includeScopeCompile: false, includeScopeProvided: false, includeScopeRuntime: false, includeSnapshotVersions: false)]) {
+                                        bat 'mvn install -pl !%DOCS% -DskipStatic=true -DskipTests=true -T 1C'
+                                        bat 'mvn clean install -B -T 1C -pl !%ITESTS% -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/%CHANGE_TARGET%'
+                                        bat 'mvn install -B -Dmaven.test.redirectTestOutputToFile=true -pl %ITESTS% -nsu'
+                                    }
                                 }
                             }
                         }
-                    }, windows: {
-                        node('proxmox-windows') {
-                            retry(3) {
-                                checkout scm
+                )
+            }
+        }
+        // The full build will be run against all regular branches
+        stage('Full Build') {
+            when { expression { env.CHANGE_ID == null } }
+            // TODO CAL-296 refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
+            steps{
+                parallel(
+                        linux: {
+                            node('linux-large') {
+                                retry(3) {
+                                    checkout scm
+                                }
+                                timeout(time: 1, unit: 'HOURS') {
```

Comment:
@oconnormi will 1 hour be enough? I've seen full builds take over 1 hour on a regular basis. Maybe that's just because of the static analysis etc...?